cmd/remote-configuration/metrics/reporter.go

```diff
@@ -0,0 +1,209 @@
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	otelresource "go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+)
+
+// MetricReporter is a metric reporter that collects Agent metrics and sends them to an
+// OTLP/HTTP destination.
+type MetricReporter struct {
+	logger types.Logger
+
+	meter           metric.Meter
+	meterShutdowner func()
+	done            chan struct{}
+
+	// The Agent's process.
+	process *process.Process
+
+	// Some example metrics to report.
+	processMemoryPhysical asyncint64.Gauge
+	counter               syncint64.Counter
+	processCpuTime        asyncfloat64.Counter
+}
+
+func NewMetricReporter(
+	logger types.Logger,
+	dest *protobufs.TelemetryConnectionSettings,
+	agentType string,
+	agentVersion string,
+	instanceId ulid.ULID,
+) (*MetricReporter, error) {
+
+	// Check the destination credentials to make sure they look like a valid OTLP/HTTP
+	// destination.
+
+	if dest.DestinationEndpoint == "" {
+		err := fmt.Errorf("metric destination must specify DestinationEndpoint")
+		return nil, err
+	}
+	u, err := url.Parse(dest.DestinationEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("invalid DestinationEndpoint: %w", err)
+	}
+
+	// Create OTLP/HTTP metric exporter.
+	opts := []otlpmetrichttp.Option{
+		otlpmetrichttp.WithEndpoint(u.Host),
+		otlpmetrichttp.WithURLPath(u.Path),
+	}
+
+	if u.Scheme == "http" {
+		opts = append(opts, otlpmetrichttp.WithInsecure())
+	}
+
+	client, err := otlpmetrichttp.New(context.Background(), opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize otlp metric http client: %w", err)
+	}
+
+	// Define the Resource to be exported with all metrics. Use OpenTelemetry semantic
+	// conventions as the OpAMP spec requires:
+	// https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#own-telemetry-reporting
+	resource, resourceErr := otelresource.New(context.Background(),
+		otelresource.WithAttributes(
+			semconv.ServiceNameKey.String(agentType),
+			semconv.ServiceVersionKey.String(agentVersion),
+			semconv.ServiceInstanceIDKey.String(instanceId.String()),
+		),
+	)
+	if resourceErr != nil {
+		return nil, resourceErr
+	}
+
+	provider := sdkmetric.NewMeterProvider(
+		sdkmetric.WithResource(resource),
+		sdkmetric.WithReader(sdkmetric.NewPeriodicReader(client)))
+
+	global.SetMeterProvider(provider)
+
+	reporter := &MetricReporter{
+		logger: logger,
+	}
+
+	reporter.done = make(chan struct{})
+
+	reporter.meter = global.Meter("opamp")
```

Comment:
```suggestion
	reporter.meter = provider.Meter("opamp")
```

No need to involve globals here.  We know which provider to use since we just created it.